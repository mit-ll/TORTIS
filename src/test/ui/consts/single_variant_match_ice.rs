enum Foo {
    Prob,
}

const FOO: u32 = match Foo::Prob { //~ ERROR unimplemented expression type
    Foo::Prob => 42,
};

const BAR: u32 = match Foo::Prob { //~ ERROR unimplemented expression type
    x => 42,
};

impl Foo {
    pub const fn as_val(&self) -> u8 {
        use self::Foo::*;

        match *self {
            //~^ ERROR loops and conditional expressions are not stable in const fn
            Prob => 0x1,
        }
    }
}

fn main() {}
