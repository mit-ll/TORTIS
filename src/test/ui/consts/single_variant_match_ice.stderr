error[E0019]: constant contains unimplemented expression type
  --> $DIR/single_variant_match_ice.rs:5:24
   |
LL | const FOO: u32 = match Foo::Prob {
   |                        ^^^^^^^^^

error[E0019]: constant contains unimplemented expression type
  --> $DIR/single_variant_match_ice.rs:9:24
   |
LL | const BAR: u32 = match Foo::Prob {
   |                        ^^^^^^^^^

error[E0723]: loops and conditional expressions are not stable in const fn
  --> $DIR/single_variant_match_ice.rs:17:15
   |
LL |         match *self {
   |               ^^^^^
   |
   = note: for more information, see issue https://github.com/rust-lang/rust/issues/57563
   = help: add `#![feature(const_fn)]` to the crate attributes to enable

error: aborting due to 3 previous errors

Some errors have detailed explanations: E0019, E0723.
For more information about an error, try `rustc --explain E0019`.
